server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregator.java

```diff
@@ -414,6 +372,74 @@ public void close() {
         }
     }
 
+    static class MultiValuesSourceCollectorImpl implements MultiTermsValuesSourceCollector {
```

Comment:
See if it's cleaner to remove static here, since aggregator itself is passed in. Use a normal inner class to access the outside aggregator. 

I tried remove `static` from here and `MultiTermsValuesSource`.

Reply:
We can use an anonymous class here but it would increase the arguments to `cartesianProductRecursive` and `apply` methods. 

Given there is a recursive method, for aggregation of 3 or more fields this would also increase the stack size. 

So, used an explicit class since anonymous class don't allow explicit constructors.

Reply:
Ignore my last comment. Will try out removing static.

Reply:
Kept static but used anonymous class to keep it similar as the previous implementation.